### DISCUSSION TITLE
calculate the local differences in dislocation density at the grain boundaries?

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#731] [Ask Anything] [answered]
Hello ,

I am sorry for bothering you again. I would like to estimate the local differences in stored energy at the grain boundaries. For this purpose, I would like to calculate the difference between KAM values of adjacent pixels on both sides of the grain boundaries (ΔKAM). I would like also to calculate ΔGAM at each grain boundary. Could you please help me to do it?

Thank you a lot for your help.

Saoussen

**Comment by ralfHielscher** [comment, marked as answer]
Hi Saoussen,

if `gB` are the grain boundary segments you are interested in, you can get the id's of the neighboring EBSD data  by
`gB.ebsdId`, i.e., 
```matlab
kam = ebsd.KAM;
dKAM = kam(gB.ebsdId(:,1)) - kam(gB.ebsdId(:,2))
```
should go into the direction you want.
Ralf.

**Comment by Saoussen-ouhiba** [comment]
Thank you so much Ralf. This is exactly what I need.

Saoussen

Le ven. 29 janv. 2021 à 05:50, Ralf Hielscher <notifications@github.com> a
écrit :
